src/simplification/SimplifyTypes.ts

```diff
@@ -81,40 +81,37 @@ function inferTypeFromValue(value: any) {
  * @param currentOutput the current output
  */
 function inferTypes(schema: Schema | boolean, currentOutput: Output) {
-  if (schema === undefined || typeof schema === 'boolean' || currentOutput === undefined) return; 
-  if (!Array.isArray(currentOutput)) return;
-  if (schema.enum) {
-    schema.enum.forEach((value: any) => {
-      const inferredType = inferTypeFromValue(value);
+  if (typeof schema === 'object' && Array.isArray(currentOutput)) {
+    if (schema.enum) {
+      schema.enum.forEach((value: any) => {
+        const inferredType = inferTypeFromValue(value);
+        if (inferredType !== undefined) {
+          addToTypes(inferredType, currentOutput);
+        }
+      });
+    }
+    //Should const overwrite the type?
+    if (schema.const !== undefined) {
+      const inferredType = inferTypeFromValue(schema.const);
       if (inferredType !== undefined) {
-        addToTypes(inferredType, currentOutput);
+        currentOutput.length = 0;
+        currentOutput.splice(0, currentOutput.length);
+        currentOutput[0] = inferredType;
       }
-    });
-  }
-  //Should const overwrite the type?
-  if (schema.const !== undefined) {
-    const inferredType = inferTypeFromValue(schema.const);
-    if (inferredType !== undefined) {
-      currentOutput.length = 0;
-      currentOutput.splice(0, currentOutput.length);
-      currentOutput[0] = inferredType;
     }
   }
 }
 
 /**
- * Infer which types the model should NOT be. 
+ * Infer which types the model should NOT be included. 
  * 
  * @param schema to go through
  * @param currentOutput the current output
  * @param seenSchemas schemas which we already have outputs for
  */
 function inferNotTypes(schema: Schema | boolean, currentOutput: Output, seenSchemas: Map<any, Output>) {
-  if (schema === undefined || currentOutput === undefined) return;
-  if (!Array.isArray(currentOutput)) return;
-  if (currentOutput.length === 0) currentOutput.push('object', 'string', 'number', 'array', 'boolean', 'null');
-  if (typeof schema === 'boolean') return;
-  if (schema.not) {
+  if (typeof schema === 'object' && schema.not && Array.isArray(currentOutput)) {
```

Comment:
I have only problem here. ^think that `schema.not` should be checked after rewrite `currentOutput` with these 7 types, like in previous code.

Reply:
Hmm, yea I can see it changes the behavior 🤔 But this is actually the correct one - "if not keyword is defined and no types have been implied, all types must be included".